Snail/Disposer.swift

```diff
@@ -8,15 +8,24 @@ public protocol DisposableType {
 
 public class Disposer {
     private(set) public var disposables: [DisposableType] = []
+    private let semaphore = DispatchSemaphore(value: 1)
```

Comment:
Instead of a semaphore, I'd go with a read / write queue. This way we ensure writes to the `disposables` array are safe without blocking the entire thread.

Take a look at: https://github.com/UrbanCompass/Snail/blob/main/Snail/Observable.swift#L10 for how we do this

Reply:
yeah that makes sense. good call